payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayLauncherActivity.kt

```diff
@@ -80,73 +86,57 @@ internal class GooglePayLauncherActivity : AppCompatActivity() {
+            }
 
-    @Deprecated("Deprecated in Java")
-    public override fun onActivityResult(
-        requestCode: Int,
-        resultCode: Int,
-        data: Intent?
-    ) {
-        super.onActivityResult(requestCode, resultCode, data)
-        if (requestCode == LOAD_PAYMENT_DATA_REQUEST_CODE) {
-            when (resultCode) {
-                RESULT_OK -> {
-                    onGooglePayResult(data)
-                }
-                RESULT_CANCELED -> {
-                    viewModel.updateResult(
-                        GooglePayLauncher.Result.Canceled
-                    )
-                }
-                AutoResolveHelper.RESULT_ERROR -> {
-                    val status = AutoResolveHelper.getStatusFromIntent(data)
-                    val statusMessage = status?.statusMessage.orEmpty()
-                    viewModel.updateResult(
-                        GooglePayLauncher.Result.Failed(
-                            RuntimeException(
-                                "Google Pay failed with error: $statusMessage"
-                            )
+            CommonStatusCodes.CANCELED -> {
+                viewModel.updateResult(
+                    GooglePayLauncher.Result.Canceled
+                )
+            }
+
+            AutoResolveHelper.RESULT_ERROR -> {
+                val statusMessage = taskResult.status.statusMessage.orEmpty()
+                viewModel.updateResult(
+                    GooglePayLauncher.Result.Failed(
+                        RuntimeException(
+                            "Google Pay failed with error: $statusMessage"
                         )
                     )
-                }
-                else -> {
-                    viewModel.updateResult(
-                        GooglePayLauncher.Result.Failed(
-                            RuntimeException(
-                                "Google Pay returned an expected result code."
-                            )
+                )
+            }
+
+            else -> {
+                viewModel.updateResult(
+                    GooglePayLauncher.Result.Failed(
+                        RuntimeException(
+                            "Google Pay returned an expected result code."
                         )
                     )
-                }
-            }
-        } else {
-            lifecycleScope.launch {
-                viewModel.onConfirmResult(
-                    requestCode,
-                    data ?: Intent()
                 )
             }
         }
     }
 
-    private fun onGooglePayResult(data: Intent?) {
-        val paymentData = data?.let { PaymentData.getFromIntent(it) }
-        if (paymentData == null) {
-            viewModel.updateResult(
-                GooglePayLauncher.Result.Failed(
-                    IllegalArgumentException("Google Pay data was not available")
-                )
+    @Deprecated("Deprecated in Java")
+    public override fun onActivityResult(
```

Comment:
Keeping the `onActivity` result method assuming it is still needed by other sections parts in this library.